### PR TITLE
WIP: Gobble last argument if it is empty

### PR DESCRIPTION
**Pull request by krlmlr**
See discussion in https://github.com/hadley/dplyr/issues/1039

This still needs some polishing -- probably most functions that accept `...` should be updated. Feedback welcome.

### CODE REVIEW

**Comment by gwerbin** [comment]
Doesn't this explicitly break something like

    in1 <- `[`
    iris %>% as.matrix %>% in1(1, )

?

**Comment by hadley** [comment]
@krlmlr I think this needs to be an option

@gwerbin magrittr doesn't use lazyeval

**Comment by krlmlr** [comment]
@hadley Like in `option(lazyeval.gobble.last.empty=TRUE)`? Can this option default to `TRUE`, please?

**Comment by hadley** [comment]
I think it would need to be a parameter to the `lazy_dots()`. Maybe `drop = TRUE` ?

**Comment by krlmlr** [comment]
How about name collisions? Should it be something like `.drop`?

**Comment by hadley** [comment]
Oh yes, `.drop` would be best.

**Comment by krlmlr** [comment]
Added argument `.drop_last_if_empty`, documentation, and a test. Needs a `document()` after merging.

What other exported functions should be enhanced?

File: R/lazy-dots.R [hunk 1]
```diff
@@ -28,12 +30,32 @@
...
+#' # this makes the terminal semicolon optional:
+#' lazy_dots(1, )
+#' lazy_dots(1)
+#' lazy_dots(.drop_last_if_empty = FALSE, 1, )
+lazy_dots <- function(..., .follow_symbols = FALSE,
+                      .drop_last_if_empty = TRUE) {
+  # Count number of arguments in ...
+  n_args <- nargs() - !missing(.follow_symbols) - !missing(.drop_last_if_empty)
+  if (n_args == 0) {
     return(structure(list(), class = "lazy_dots"))
```
Comment (krlmlr) [inline R/lazy-dots.R]: If I remove this shortcut, the test runs indefinitely on my machine. Still, the code would look cleaner without the special treatment of an empty argument list.

Comment (hadley) [inline R/lazy-dots.R]: Might be better to fix in C code?

Comment (krlmlr) [inline R/lazy-dots.R]: Sounds good. I can give it a try later this week, but I don't mind if you (or anybody else) take this.

**Pull request closed without merging**